### PR TITLE
FEAT-#26: setup of database and start of routes apis

### PR DESCRIPTION
**Pull request by Sebastien0730**
# Description

Brief summary of the changes made.

Issue reference: #26 

## Branch and Commit name must follow this template
```
Branch template: {ISSUE TYPE}-{Issue number}/{desciption_of_branch}
Commit template: {ISSUE TYPE}-{Issue number}: {description of changes}
```

## Type of change

Please delete options that are not relevant.

- [ ] BUG (Fixing or documenting a bug)
- [x] FEAT (Add a new feature to the project)
- [ ] CHORE (Something that needs to be done outside of the issue)
- [ ] DOCS (Implement documentation)
- [ ] TASK (A task that must be done to complete an issue)

# Checklist:

- [x] My pull request references the issue in the comments
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Test

- [ ] Ran `make test-unit` and all test passes
- [ ] Ran `make test-coverage` and coverage is atleast 90%

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,84 @@
...
+robotframework==3.1.2
+scipy==1.3.1
+selenium==3.141.0
+signalrcore==0.7.9
+six==1.12.0
+soupsieve==1.9.2
+sqlparse==0.3.0
+tensorboard==1.14.0
+tensorflow==1.14.0
+tensorflow-estimator==1.14.0
```
Comment (phong1233) [inline backend/requirements.txt]: I believe that using tensorflow for our project is currently out of the scope

File: backend/database/models.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# User, picture,
+
+
+class User(db.Document):
+    email = db.EmailField(required=True, unique=True)
+    username = db.StringField(required=True, unique=True)
+    fname = db.StringField(required=True)
+    lname = db.StringField()
+    password = db.StringField(required=True)
+    own_pictures = db.ListField()
```
Comment (phong1233) [inline backend/database/models.py]: pictures

File: backend/database/models.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    liked_pictures = db.ListField()
```
Comment (phong1233) [inline backend/database/models.py]: This should go in the picture schema

File: backend/requirements.txt (hunk 1 above)
Comment (Sebastien0730) [reply backend/requirements.txt]: Sorry it was added automatically, I removed it in last commit

File: backend/database/models.py (hunk 2 above)
Comment (Sebastien0730) [reply backend/database/models.py]: changed it

File: backend/database/models.py (hunk 3 above)
Comment (Sebastien0730) [reply backend/database/models.py]: I changed a bit what is here. Let's talk about that at the next meeting

File: backend/requirements.txt [hunk 4]
```diff
@@ -1,3 +1,68 @@
...
+idna==2.8
+imageio==2.5.0
+isort==4.3.4
+itsdangerous==1.1.0
+Jinja2==2.10.3
+Keras-Applications==1.0.8
+Keras-Preprocessing==1.1.0
+keyboard==0.13.3
+kiwisolver==1.1.0
+lazy-object-proxy==1.3.1
```
Comment (phong1233) [inline backend/requirements.txt]: We dont need this

File: backend/database/models.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
+from .db import db
+
+# Create a class for all table that should be in the database
+
+
+class User(db.Document):
```
Comment (phong1233) [inline backend/database/models.py]: I think we should add the schema in the doc folder. For example:
```json
"user" : {
  "username": "Sebastien",
  "pictures": [
    "pic1",
    "pic2"
  ]
}
```

**Pull request merged**